contracts/src/Gateway.sol

```diff
@@ -220,8 +224,8 @@ contract Gateway is IGateway, IInitializable {
     }
 
     function agentOf(bytes32 agentID) external view returns (address) {
-        address agentAddress = _ensureAgent(agentID);
-        return agentAddress;
+        // No error checking here, because the agent will not exist at first and needs to be created.
```

Comment:
I would rather document the behavior in a more generic manner in a doc comment above the function. For example:

> A zero address is returned if no matching agent contract is found

This will also help to convey to auditors that this an explicit design choice on our part.

Reply:
I am not sure if we need to make this change in `agentOf` to stop the relayers from restarting. We should only need this for `channelNoncesOf`. Another thing we could do is not change the contracts at all and change the relayers to ignore the specific error, log it out and return default `(0, 0)` for nonces.

Reply:
Agree. I would assume the restarting behavior as expected and not worthwhile to change contract just for that. 

Another option is to revamp the e2e script a bit to first create agent/channel before starting the template relayer.  To not block the e2e setup too long we can move these to a seperate script.  But anyway, it's just a `nice to have` improvement and not essential.

Reply:
> Another thing we could do is not change the contracts at all and change the relayers to ignore the specific error, log it out and return default (0, 0) for nonces

Yes, I like this idea. It means we can leave the contracts alone. 👍 

Otherwise we'd have to update all four getters in this file to stop using `_ensure*`.